lib/src/lib.rs

```diff
@@ -75,11 +79,41 @@ struct DeployedActor {
     address: Address,
 }
 
-/// An Actor that has been deployed into a `BlockStore`.
+/// Outcome of the test.
+#[derive(Clone, Debug)]
+pub enum TestResultType {
+    Passed(ApplyRet),
+    Failed(ApplyRet),
+    /// TODO: Upgrade to a proper `Reason` struct?
+    /// We Receive `anyhow::Result` from upstream so, there's probably
+    // not much we can do.
+    Erred(String),
```

Comment:
Wouldn't a `reason` information live at the same level as `Failed` or `Erred` ?

Reply:
no, `Erred` is the consequence of `execute_message` failing, so we have a source `Error`. `Failed` is consequence of the exit_code being different than the expected one for the `Method` right?